flask_monitoringdashboard/database/request.py

```diff
@@ -57,6 +57,7 @@ def add_request(db_session, duration, endpoint_id, ip, group_by, status_code):
     )
     db_session.add(request)
     db_session.flush()
+    db_session.commit()
```

Comment:
Based on the error message, your analysis seems correct. We actually added at some point `db_session.flush()` to fix precisely this problem, but apparently it wasn't enough. 

Looking at this [post](https://medium.com/@oba2311/sqlalchemy-whats-the-difference-between-a-flush-and-commit-baec6c2410a9) and this [SO answer](https://stackoverflow.com/a/4202016), it appears that `commit()` always calls `flush()`, so `db_session.flush()` should be removed since it's redundant.